src/AWS.Deploy.CLI/Commands/DeployCommand.cs

```diff
@@ -317,6 +320,41 @@ private async Task<Recommendation> GetSelectedRecommendationFromPreviousDeployme
             return selectedRecommendation;
         }
 
+        private async Task<Recommendation?> GetRecommendationForRedeployment(List<Recommendation> recommendations, CloudApplication deployedApplication, string deploymentProjectPath)
+        {
+            var targetRecipeId = !string.IsNullOrEmpty(deploymentProjectPath) ?
+                await GetDeploymentProjectRecipeId(deploymentProjectPath) : deployedApplication.RecipeId;
+
+            foreach (var recommendation in recommendations)
+            {
+                if (string.Equals(recommendation.Recipe.Id, targetRecipeId) && _deployedApplicationQueryer.IsCompatible(deployedApplication, recommendation))
+                {
+                    return recommendation;
+                }
+            }
+            return null;
+        }
+
+        private async Task<string> GetDeploymentProjectRecipeId(string deploymentProjectPath)
+        {
+            if (!_directoryManager.Exists(deploymentProjectPath))
+            {
+                throw new InvalidOperationException($"Invalid deployment project path. {deploymentProjectPath} does not exist on the file system.");
+            }
+            try
+            {
+                var recipeFileName = _directoryManager.GetDirectoryInfo(deploymentProjectPath).Name + ".recipe";

```

Comment:
This is making the assumption the recipe file is the same name as the directory. We don't have that rule. Search the directory for the recipe file.

Reply:
Fixed